### PR TITLE
[SC-765] Combine ETH orders mappings into one

### PR DESCRIPTION
**Pull request by typicalbuster**

### CODE REVIEW

File: contracts/helpers/ETHOrders.sol [hunk 1]
```diff
@@ -23,12 +23,12 @@ contract ETHOrders is IPostInteraction, OnlyWethReceiver {
...
+    uint256 constant private _BALANCE_BIT_SHIFT = 160;
+
     address private immutable _limitOrderProtocol;
     IWETH private immutable _WETH; // solhint-disable-line var-name-mixedcase
-    // Tracks ETH balances for orders.
-    mapping(bytes32 => uint256) public ethOrderBalance;
-    // Order makers to verify signatures against
-    mapping(bytes32 => address) public orderMakers;
+    /// @notice Makers and their uint96 ETH balances in single mapping.
+    mapping(bytes32 => uint256) private _ordersMakersBalances;
```
Comment (k06a) [inline contracts/helpers/ETHOrders.sol:31]: Let’s use struct with address and uint96, code will be cleaner

**Review by k06a — CHANGES_REQUESTED** [review]

**Comment by k06a** [comment]
Let’s also have one getter returning two arrays

File: contracts/helpers/ETHOrders.sol [hunk 2]
```diff
@@ -68,21 +64,20 @@ contract ETHOrders is IPostInteraction, OnlyWethReceiver {
         bytes calldata interaction = extension.postInteractionTargetAndData();
         if (interaction.length != 20 || address(bytes20(interaction)) != address(this)) revert InvalidOrder();
         orderHash = IOrderMixin(_limitOrderProtocol).hashOrder(order);
-        if (orderMakers[orderHash] != address(0)) revert ExistingOrder();
-        orderMakers[orderHash] = msg.sender;
-        ethOrderBalance[orderHash] = msg.value;
+        if (ordersMakersBalances[orderHash].maker != address(0)) revert ExistingOrder();
+        ordersMakersBalances[orderHash] = ETHOrder(msg.sender, uint96(msg.value));
```
Comment (k06a) [inline contracts/helpers/ETHOrders.sol:68]:
```suggestion
        ordersMakersBalances[orderHash] = ETHOrder({
            maker: msg.sender,
            balance: uint96(msg.value)
        });
```

File: contracts/helpers/ETHOrders.sol [hunk 3]
```diff
@@ -68,21 +64,20 @@ contract ETHOrders is IPostInteraction, OnlyWethReceiver {
...
         _WETH.safeDeposit(msg.value);
         emit ETHDeposited(orderHash, msg.value);
     }
 
     /**
-     * @notice Sets ethOrderBalance to 0, refunds ETH and does standard order cancellation on Limit Order Protocol.
+     * @notice Sets _ordersMakersBalances to 0, refunds ETH and does standard order cancellation on Limit Order Protocol.
```
Comment (k06a) [inline contracts/helpers/ETHOrders.sol:74]:
```suggestion
     * @notice Sets `ordersMakersBalances` to 0, refunds ETH and does standard order cancellation on Limit Order Protocol.
```

**Review by ZumZoom — APPROVED** [review]

**Pull request merged**